### PR TITLE
feat (driver) implement our own LED driver

### LINKED ISSUE #66: Bug: Cache disabled but cached memory region accessed
I recently experienced different, randomly appearing crashes like:

```
Guru Meditation Error: Core  1 panic'ed (Cache disabled but cached memory region accessed). 

Core  1 register dump:
PC      : 0x4011b968  PS      : 0x00060035  A0      : 0x800814ae  A1      : 0x3ffbf1fc
A2      : 0x00000002  A3      : 0x00000001  A4      : 0xbad00bad  A5      : 0x00000000  
A6      : 0x00000006  A7      : 0x00000000  A8      : 0x8011b963  A9      : 0x3ffbf1dc  
A10     : 0x00000001  A11     : 0xffffffff  A12     : 0x000f8023  A13     : 0x00000006  
A14     : 0x007bf2a8  A15     : 0x00000000  SAR     : 0x0000000f  EXCCAUSE: 0x00000007
EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000  


Backtrace:0x4011b965:0x3ffbf1fc |<-CORRUPTED

  #0  0x4011b965:0x3ffbf1fc in xPortEnterCriticalTimeoutSafe at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/freertos/port/xtensa/include/freertos/portmacro.h:590
      (inlined by) vPortEnterCriticalSafe at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/freertos/port/xtensa/include/freertos/portmacro.h:599
      (inlined by) rmt_set_tx_intr_en at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/driver/rmt.c:497
```

```
Guru Meditation Error: Core  1 panic'ed (Cache disabled but cached memory region accessed). 

Core  1 register dump:
PC      : 0x4011bae0  PS      : 0x00060035  A0      : 0x800817f2  A1      : 0x3ffbf22c  
A2      : 0x3ffc3220  A3      : 0x00000002  A4      : 0x0ffd114c  A5      : 0xfffffffe  
A6      : 0x0000001a  A7      : 0x00000008  A8      : 0x800813d5  A9      : 0x00000008  
A10     : 0x00000002  A11     : 0x00000000  A12     : 0x00000010  A13     : 0x00000000  
A14     : 0x00000003  A15     : 0x00060323  SAR     : 0x00000012  EXCCAUSE: 0x00000007  
EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000  


Backtrace:0x4011badd:0x3ffbf22c |<-CORRUPTED

  #0  0x4011badd:0x3ffbf22c in rmt_set_tx_thr_intr_en at ??:?
```

```
Guru Meditation Error: Core  1 panic'ed (Cache disabled but cached memory region accessed). 

Core  1 register dump:
PC      : 0x4011b92c  PS      : 0x00060035  A0      : 0x800817f2  A1      : 0x3ffbf22c
A2      : 0x3ffc3080  A3      : 0x0028800a  A4      : 0x0ffd114c  A5      : 0xfffffffe  
A6      : 0x0000001a  A7      : 0x00000008  A8      : 0x800814ae  A9      : 0x3ffbf20c  
A10     : 0x00000002  A11     : 0x00000001  A12     : 0x3ff56ac8  A13     : 0x00000000  
A14     : 0x00000006  A15     : 0x00000000  SAR     : 0x00000007  EXCCAUSE: 0x00000007  
EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000  


Backtrace:0x4011b929:0x3ffbf22c |<-CORRUPTED

  #0  0x4011b929:0x3ffbf22c in rmt_module_enable at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/driver/rmt.c:130
```

I have no clue at the moment, why this is happening, what exactly is causing it and how to fix it. But we need to do something about this otherwise I see the project at risk. ~~Unfortunately, I can't provide any code that is constantly reproducing it.~~ [click](https://github.com/TheRealKasumi/TesLight/tree/bug/cache-disable-but-chached-memory-region-accessed). Using WiFi in `STA` mode seems to trigger it more frequently. Same for using the `Os` optimization flag. Also, it seems to be related to the `rmt` driver. At least it can be found in almost every stack trace.

Maybe related, but should be fixed in the meantime: https://github.com/espressif/esp-idf/issues/9487

### LINKED ISSUE #98: Bug: Stale LEDs when reducing LED count
**Describe the bug**
When reducing the LED count in the UI below the number of LEDs on the strip, then the "extra" LEDs remain at their last status instead of being turned off.

**To Reproduce**
1. Turn on the light on any zone, make sure all LEDs are lit up
3. Go to 'Advanced' in the zone configuration and reduce the 'LED Count' to a number that is smaller the the actual count of LEDs in that zone.
4. Click 'Apply Settings'
5. Observe that the LEDs above the LED count are still lighting.

**Expected behavior**
When changing the LED Count any LED on the strip that is above the count should be turned off.

**Additional context**
The 'Average Power' that is printed on the console is also affected by this, as it only calculates the power based on the configured LEDs, not the actually lit up LEDs

### LINKED ISSUE #134: Animation stuttering every ~500ms
I assume this happens when reading the temperature sensors since they are quite slow and blocking the controller. To be checked.

### LINKED ISSUE #140: Create own LED driver and remove FastLED lib
FastLED brought us a lot of random bugs, errors and crashes that do not happen without it. So I decided to write down our own LED driver.

- [x] Use I2S
- [x] Should be memory efficient
- [x] Should be fast and drive 1600 (200 per zone) without a problem 
- [x] Check stability and make sure its safe to use with webserver and multiple tasks

### PR DESCRIPTION
**Pull request by TheRealKasumi**
closes #140
closes #98 
partially closes #134

Its finally here, our own driver for WS2812(B) LEDs. With that we remove the FastLED dependency and potentially a lot of problems which startet when we included it. The performance with the new driver has roughly doubled while the memory consumption was also slightly reduced. The CPU load also reduced drastically since we are using DMA with the I2S hardware. So far it seems perfectly stable but I am still in the process of testing. There might be smaller improvements to the way how pixels are accessed but this is more about... "making it look nicer" while not changing anything functional.

Also note that this potentially seems to close #66. So if we don't encounter this again I would guess we close it in the future. Alright, I guess all this is pretty good news.

### CODE REVIEW

**Review by JakobLichterfeld — APPROVED** [review]
lgtm, really good :-) Thank you for your extraordinary efforts!

File: mcu/platformio.ini [hunk 1]
```diff
@@ -14,14 +14,13 @@ board = az-delivery-devkit-v4
 framework = arduino
 board_build.partitions = partition.csv
 build_type = release
-build_flags = -O2 ; Better don't use O3, partially breaks code and even hits ICEs
+build_flags = -O3
```
[the remark this answers is not in the document]
Comment (TheRealKasumi) [reply mcu/platformio.ini:17]: Hahaha, at least it seems to be working now :P . However I would vote for keeping this thing open for a few day. In theory its all fine and working now but before doing this big change I would love to do some more testing. You know... not to stumble over random things again.

**Comment by TheRealKasumi** [comment]
Tested the current code for 24h now. I assume its stable. So I would say we can merge it after you approved.

**Review by JakobLichterfeld — APPROVED** [review]
lgtm. I could not test it atm, I need to get another power supply first...

**Comment by TheRealKasumi** [comment]
If this happens in the near future we can also wait😋. Having a second tester would be great.

**Comment by JakobLichterfeld** [comment]
> I need to get != I get ... :-)

**Comment by TheRealKasumi** [comment]
Alright, then I am going for the merge now😜.

**Pull request merged**